addons/mail/static/src/models/emoji_registry.js

```diff
@@ -1,110 +1,44 @@
 /** @odoo-module **/
 
 import { registerModel } from '@mail/model/model_core';
-import { many } from '@mail/model/model_field';
-import { insertAndReplace } from '@mail/model/model_field_command';
+import { attr, many } from '@mail/model/model_field';
+import { insert } from '@mail/model/model_field_command';
+import { emojiData } from '@mail/models_data/emoji_data';
 
 registerModel({
     name: 'EmojiRegistry',
     identifyingFields: ['messaging'],
+    lifecycleHooks: {
+        _created() {
+            this._populateFromEmojiData();
+        },
+    },
+    recordMethods: {
+        _populateFromEmojiData() {
+            this.models['Emoji'].insert(emojiData.map(emoji => {
+                return {
+                    unicode: emoji.codepoints,
+                    sources: [...emoji.shortcodes, ...emoji.emoticons],
+                    keywords: [...emoji.shortcodes, ...emoji.emoticons, ...emoji.name, ...emoji.keywords],
+                    description: emoji.name,
+                    emojiCategories: insert([
+                        { categoryName: "all" },
+                        { categoryName: emoji.category },
+                    ]),
+                    hasSkinToneVariations: emoji.hasSkinToneVariations,
+                };
+            }));
```

Comment:
I think the script should also populate explicitly the categories, and categories should have data that determine their sequence, so that there's proper `sort` of list of categories.